strawberryfields/api/result.py

```diff
@@ -107,6 +103,70 @@ def state(self):
             raise AttributeError("The state is undefined for a stateless computation.")
         return self._state
 
+    @staticmethod
+    def combine_samples(samples_list, mode_order):
+        """Combine samples measured at different times into one nested array.
+
+        Args:
+            samples_list (list[array, tensor]): the sample measurements in a list
+            mode_order (list[array]): The nested mode order for the measured modes. Must have the
+                same structure as samples_list.
+
+        Returns:
+            array: the samples in ascending mode order with shape ``(shots, measured_modes)``
+        """
+
+        shots = len(samples_list[0])
+
+        mode_order_flat = np.hstack(mode_order)
+        len_modes = len(set(mode_order_flat))
+
+        # remove duplicate mode-measurements and only return the modes from the last measurement
+        if len(mode_order_flat) != len_modes:
```

Comment:
(Minor suggestion): could help readability
```suggestion
        duplicate_mode_measurements = len(mode_order_flat) != len_modes
        if duplicate_mode_measurements:
```

Reply:
I like this. Makes it a bit clearer why this is done. :+1: